cmd/sti/main.go

```diff
@@ -54,19 +59,35 @@ func newCmdVersion() *cobra.Command {
 }
 
 func newCmdBuild(req *api.Request) *cobra.Command {
+	useConfig := false
+
 	buildCmd := &cobra.Command{
 		Use:   "build <source> <image> <tag>",
 		Short: "Build a new image",
 		Long:  "Build a new Docker image named <tag> from a source repository and base image.",
 		Run: func(cmd *cobra.Command, args []string) {
-			if len(args) < 3 {
+			// Attempt to restore the build command from the configuration file
+			if useConfig {
+				config.Restore(req, cmd)
+			}
+
+			// If user specifies the arguments, then we override the stored ones
```

Comment:
Actually I don't think it should be part of Restore() as the `use-config` will have to be parsed in Restore as well and by default the user-config is disabled. In that case setting the args will be hidden in Restore()...